### PR TITLE
feat: docker-compose file with infura support

### PR DESCRIPTION
**Pull request by unnawut**
:clipboard: #1029

## Overview

Allows docker-compose up a local environment with infura instead of a local geth.

## Changes

- Added `docker-compose-infura.yml`
- Added `make docker-start-cluster-with-infura` that runs `docker-compose ... up` with infura settings
- Prettified `make help`
- Git-ignored `docker-compose.override.yml`
- Renamed `plasma-deployer` service name to `plasma-contracts` to reflect the current repo

## Testing / Usage

1. Configure the missing values in `docker-compose-infura.yml`
2. Run `make docker-start-cluster-with-infura` or directly with `docker-compose -f docker-compose.yml -f docker-compose-infura.yml up`
3. Voila

Since you would want to exclude your infura and private keys from the repo, feel free to copy over `docker-compose-infura.yml` to `docker-compose.override.yml` then make the changes, which will be ignore by git. Then you can simply run `make docker-start-cluster` or `docker-compose up` since the override file is natively and automatically loaded by docker-compose.

## Merge blockers
- https://github.com/omisego/plasma-deployer/pull/4 (the deployer's docker image tag after merge must be updated in this PR)
- https://github.com/omisego/plasma-contracts/pull/421 (`--network remote`)

### CODE REVIEW

**Comment by achiurizo** [comment]
@unnawut can we go ahead and merge this?

**Comment by unnawut** [comment]
@achiurizo Waiting for https://github.com/omisego/plasma-deployer/pull/4 approval then this can be merged.

File: docker-compose-infura.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+version: "2.1"
+services:
+  plasma-deployer:
```
Comment (InoMurko) [inline docker-compose-infura.yml]: now that we don't use plasma-deployer in standard `docker-compose` this won't get overwritten properly, I think we need to bring the full plasma-deployer description here, referencing the tag that supports infura

Comment (unnawut) [reply docker-compose-infura.yml]: @InoMurko It seems it should be possible to follow the latest `docker-compose.yml` way of deploying the contracts for infura too. Any red flags if I follow that route? It'll simplify things a lot not having to deploy contracts in different ways.

Comment (InoMurko) [reply docker-compose-infura.yml]: nop, not really! I would love to see us deprecating plasma-deployer completely!

**Comment by unnawut** [comment]
@InoMurko Updated. Ready for another review.

File: docker-compose.yml [hunk 2]
```diff
@@ -15,17 +15,17 @@ services:
       timeout: 3s
       retries: 5
 
-  plasma-deployer:
+  plasma-contracts:
```
Comment (InoMurko) [inline docker-compose.yml]: good decision!

**Comment by InoMurko** [comment]
To me this seems fine, but I wonder why @kendricktan decided to use infura values initially. Let's wait for his review as well!

**Comment by unnawut** [comment]
He mentioned:

> Its just how the truffle-config.js is setup, we should probably rename infura to remote_provider or something more generic so it makes more sense

And so I made this PR https://github.com/omisego/plasma-contracts/pull/421 which should be merged & docker image built before merging this PR.

**Comment by unnawut** [comment]
Pending https://github.com/omisego/plasma-contracts/pull/421 merge

**Comment by unnawut** [comment]
Turns out this PR is blocked by https://github.com/omisego/elixir-omg/blob/master/plasma-contract.Dockerfile#L15 referring to a relatively old commit of `plasma-contracts`.

To unblock, with Ino's guidance:
- Bump [plasma-contract.Dockerfile](https://github.com/omisego/elixir-omg/blob/master/plasma-contract.Dockerfile#L15) to https://github.com/omisego/plasma-contracts/commit/a6d870f4562b359bece7bea01643e4a6b567b5a9 or later
- Bump [mix.lock](https://github.com/omisego/elixir-omg/blob/master/mix.lock#L58) also to https://github.com/omisego/plasma-contracts/commit/a6d870f4562b359bece7bea01643e4a6b567b5a9 or later

While not blocking this PR, the update above will also need to be reflected on to:
- [`docker-elixir-omg`'s #4](https://github.com/omisego-images/docker-elixir-omg/pull/4/files#diff-d19bed7c9bcebdb3338f79268b811cc1R15), get merged, image built, and pushed to Docker Hub
- Then bump [`elixir-omg`'s #1093](https://github.com/omisego/elixir-omg/pull/1093/files#diff-4e5e90c6228fd48698d074241c2ba760R19) to use the latest `docker-elixir-omg` image tag

cc: @nicholasmueller since you're also facing this same situation with omg-js

**Comment by unnawut** [comment]
Blocked by https://github.com/omisego/elixir-omg/pull/1093 and https://github.com/omisego/elixir-omg/pull/1100

**Comment by unnawut** [comment]
Will merge this when I wake up so I'm availble for support.

**Pull request merged**